src/bdc/steps/helpers/generate_hash_leads.py

```diff
@@ -0,0 +1,80 @@
+# SPDX-License-Identifier: MIT
+# SPDX-FileCopyrightText: 2023 Ruchita Nathani <ruchita.nathani@fau.de>
+
+import hashlib
+import os
+from datetime import datetime
+
+import pandas as pd
+
+from bdc.steps.step import Step
+from database import get_database
+from logger import get_logger
+
+log = get_logger()
+
+
+class LeadHashGenerator:
+    BASE_PATH = os.path.dirname(__file__)
+    _curr_lookup_table = ("", None)
+
+    def hash_lead(self, lead_data):
```

Comment:
One suggestion to improve efficiency would be to move the calculation of hashes into a dedicated step where the hashes are added to the main dataframe. Otherwise the hashes would be calculated for every lead in every step again.

Reply:
Good point this was added